vega_sim/api/data_raw.py

```diff
@@ -100,30 +100,31 @@ def asset_info(
     ).asset
 
 
-def all_market_accounts(
-    asset_id: str,
-    market_id: str,
+def list_accounts(
     data_client: vac.VegaTradingDataClientV2,
-) -> Dict[vega_protos.vega.AccountType, MarketAccount]:
+    asset_id: Optional[str] = None,
+    market_id: Optional[str] = None,
+    party_id: Optional[str] = None,
+) -> List[data_node_protos_v2.trading_data.AccountBalance]:
     """
     Output liquidity fee account/ insurance pool in the market
     """
 
-    account_filter = data_node_protos_v2.trading_data.AccountFilter(
-        asset_id=asset_id,
-        market_ids=[market_id],
-    )
-
-    accounts = unroll_v2_pagination(
+    account_filter = data_node_protos_v2.trading_data.AccountFilter()
+    if party_id is not None:
+        account_filter.party_ids.extend([party_id])
+    if asset_id is not None:
+        account_filter.asset_id = asset_id
+    if market_id is not None:
+        account_filter.market_ids.extend([market_id])
+    return unroll_v2_pagination(
         base_request=data_node_protos_v2.trading_data.ListAccountsRequest(
             filter=account_filter
         ),
         request_func=lambda x: data_client.ListAccounts(x).accounts,
         extraction_func=lambda res: [i.account for i in res.edges],
```

Comment:
Not an issue with this PR, just a heads up this is going to break with vega > [v0.63.0](https://github.com/vegaprotocol/vega/releases/tag/v0.63.0) (Issue [#6275](https://github.com/vegaprotocol/vega/issues/6725) specifically)

Can't remember when we update our vega tag in market-sim but maybe worth doing as part of this PR?

Reply:
👌

Reply:
Need to attempt the wallet v2 update too so I guess I'll bundle it all up together